src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -80,6 +88,34 @@ object TypeCast {
     }
   }
 
+  @scala.annotation.tailrec
+  def parseXmlDate(formatters: List[DateTimeFormatter], value: String): Date = {
```

Comment:
I don't mind this construction with `tailrec`; I usually just write

```
for (formatter <- formatters) {
  try {
    return ...valueOf...
  } catch {
    _: Exception => // continue
  }
}
throw new IllegalArgumentException(...)
```

Reply:
This is a personal choice and as you are the maintainer I am happy to change to your style if you would like?

Reply:
Eh, I don't feel strongly about it. Won't matter. I find the for loop simpler but I get the Scala construct here